workload/schemachange: check enum's non-public status during ADD REGION

otan opened the pull request:
Previously, having two ADD REGIONS in the same transaction with the same
value would error "unexpectedly" by the workload changer as we did not
inject the expected error in. This is because the region on the first
ADD REGION is not immediately made a public enum member, and does not
show up on SHOW REGIONS on the second generation of ADD REGION.

To fix this, we now detect whether the enum exists as a non-public member
and if so add the expected error code.

Also re-enables ADD REGION operation generation.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/62705)
<!-- Reviewable:end -->

otan committed:
workload/schemachange: check enum's non-public status during ADD REGION

Previously, having two ADD REGIONS in the same transaction with the same
value would error "unexpectedly" by the workload changer as we did not
inject the expected error in. This is because the region on the first
ADD REGION is not immediately made a public enum member, and does not
show up on SHOW REGIONS on the second generation of ADD REGION.

To fix this, we now detect whether the enum exists as a non-public member
and if so add the expected error code.

Also re-enables ADD REGION operation generation.

Release note: None

ajwerner reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 2 of 2 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/62705#-:-MWxpkOkBjX6pINX66fd:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @otan)



<!-- Sent from Reviewable.io -->

otan commented:
bors r=ajwerner
